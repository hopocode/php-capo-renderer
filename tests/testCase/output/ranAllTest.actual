<nav><ul class="top-nav"><li><a href="/">Home</a></li><li><a href="/about-us">&lt;b&gt;About us&lt;/b&gt;</a></li><li><a href="/support"><i>support</i></a></li><li><a href="/support" class="spt"><i>support</i></a></li></ul></nav>