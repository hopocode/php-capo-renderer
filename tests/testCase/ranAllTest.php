<?php

declare(strict_types=1);

use Hopo\PhpCopo\CapoRender;
use Tester\Assert;

require __DIR__ . '/../boot.php';

// $spec = CapoRender::specFileToHtmlString("https://raw.githubusercontent.com/hopocode/component-renderer-out/main/src/hello-world.json");
// $html = file_get_contents("https://raw.githubusercontent.com/hopocode/component-renderer-out/main/src/hello-world.html");
// Assert::same($html, $spec);

// $spec = CapoRender::specFileToHtmlString("https://raw.githubusercontent.com/hopocode/component-renderer-out/main/src/html-escape.json");
// $html = file_get_contents("https://raw.githubusercontent.com/hopocode/component-renderer-out/main/src/html-escape.html");
// Assert::same($html, $spec);

$spec = CapoRender::specFileToHtmlString("https://raw.githubusercontent.com/hopocode/component-renderer-out/main/src/simple-menu.json");
$html = file_get_contents("https://raw.githubusercontent.com/hopocode/component-renderer-out/main/src/simple-menu.html");
function sanitize_output($buffer) {

    $search = array(
        '/\>[^\S ]+/s',     // strip whitespaces after tags, except space
        '/[^\S ]+\</s',     // strip whitespaces before tags, except space
        '/(\s)+/s',         // shorten multiple whitespace sequences
        '/<!--(.|\s)*?-->/' // Remove HTML comments
    );

    $replace = array(
        '>',
        '<',
        '\\1',
        ''
    );

    $buffer = preg_replace($search, $replace, $buffer);
    $b = strtr($buffer, ["> <" => "><"]);
    return $b;
}



$htmlNoSpace = sanitize_output($html);
Assert::same($htmlNoSpace, $spec);



