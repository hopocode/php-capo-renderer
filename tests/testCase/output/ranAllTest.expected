<nav><ul class="top-nav"><li><a href="/">Home</a></li><li><a href="/about-us"><b>About us</b></a></li><li><a class="spt" href="/support"><i>support</i></a></li></ul></nav>